pkg/app/mysql.go

```diff
@@ -0,0 +1,244 @@
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type chartInfo struct {
+	helmrepoURL   string
+	helmChartName string
+	helmAppname   string
+}
+
+type MysqlDB struct {
+	cli       kubernetes.Interface
+	namespace string
+	password  string
+	accessID  string
+	secretKey string
+	region    string
+	chart     chartInfo
+}
+
+func NewMysqlDB(helmrepoURL, helmChart, helmAppname string) App {
+	return &MysqlDB{
+		password: "mysecretpassword",
+		chart: chartInfo{
+			helmrepoURL:   helmrepoURL,
+			helmChartName: helmChart,
+			helmAppname:   helmAppname,
+		},
+	}
+}
+
+func (mdb *MysqlDB) Init(ctx context.Context) error {
+	var ok bool
+	mdb.region, ok = os.LookupEnv(awsconfig.Region)
+	if !ok {
+		return fmt.Errorf("Env var %s is not set", awsconfig.Region)
+	}
+	mdb.accessID, ok = os.LookupEnv(awsconfig.AccessKeyID)
+	if !ok {
+		return fmt.Errorf("Env var %s is not set", awsconfig.AccessKeyID)
+	}
+
+	mdb.secretKey, ok = os.LookupEnv(awsconfig.SecretAccessKey)
+	if !ok {
+		return fmt.Errorf("Env var %s is not set", awsconfig.SecretAccessKey)
+	}
+
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return nil
+	}
+
+	mdb.cli, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (mdb *MysqlDB) Install(ctx context.Context, namespace string) error {
+
+	mdb.namespace = namespace
+
+	cli := testutil.NewHelmClient()
+	log.Info("Adding repo")
+	err := cli.AddRepo(ctx, mdb.chart.helmChartName, mdb.chart.helmrepoURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Info("MysqlDB: Installing mysql instance using helm.")
+	err = cli.Install(ctx, mdb.chart.helmChartName+"/"+mdb.chart.helmAppname, "mysql-test", mdb.namespace, map[string]string{"mysqlRootPassword": mdb.password, "persistence.enabled": "false", "imagePullPolicy": "Always"})
+	if err != nil {
+
+		log.Fatal(err)
+	}
+
+	return nil
+}
+
+func (mdb *MysqlDB) IsReady(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	defer cancel()
+	err := kube.WaitOnDeploymentReady(ctx, mdb.cli, mdb.namespace, "mysql-test")
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+
+}
+
+func (mdb *MysqlDB) Object() crv1alpha1.ObjectReference {
+
+	return crv1alpha1.ObjectReference{
+		Kind:      "deployment",
+		Name:      "mysql-test",
+		Namespace: mdb.namespace,
+	}
+}
+
+func (mdb *MysqlDB) Uninstall(ctx context.Context) error {
+	log.Info("MysqlDB: UnInstalling mysql instance.")
+
+	cli := testutil.NewHelmClient()
+
+	err := cli.Uninstall(ctx, "mysql-test", mdb.namespace)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return nil
+}
+
+func (mdb *MysqlDB) Ping(context.Context) error {
+
+	podname, containername, err := mdb.GetPodAndContainerName(mdb.namespace)
+
+	// exec into the pod and create the test database, read password from secret
+	loginMysql := []string{"sh", "-c", "mysql -u root --password=" + mdb.password + ""}
```

Comment:
Is it possible to use database/SQL and the fo driver?

Reply:
Hi @tdmanv ,
I think to use database/SQL pkg and the driver we will have to expose the mysql deployment either using service (NodePort) or ingress I think.
What do you suggest ?

Reply:
Hi @tdmanv 
as discussed lets go ahead with execing into the pod to talk to the application.